crates/trigger/src/loader.rs

```diff
@@ -49,18 +49,23 @@ impl Loader for TriggerLoader {
             .as_ref()
             .context("LockedComponentSource missing source field")?;
         let path = parse_file_url(source)?;
-        spin_core::Component::new(
-            engine,
-            spin_componentize::componentize_if_necessary(&fs::read(&path).await.with_context(
-                || {
-                    format!(
-                        "failed to read component source from disk at path '{}'",
-                        path.display()
-                    )
-                },
-            )?)?,
-        )
-        .with_context(|| format!("loading module {path:?}"))
+        let bytes = fs::read(&path).await.with_context(|| {
+            format!(
+                "failed to read component source from disk at path '{}'",
+                path.display()
+            )
+        })?;
+        let component = spin_componentize::componentize_if_necessary(&bytes)?;
+        let was_already_component = matches!(component, std::borrow::Cow::Borrowed(_));
```

Comment:
This feels like a very fragile dependency on an implementation detail of `componentize_if_necessary`.  Would it be prohibitive to modify `spin-componentize` to return "was it originally a component or module" as well as the componentized bytes?

Reply:
I'm sort of disagree that it's fragile as it is directly tied to the fact that the bytes that go into `componentize_if_necessary` are the same that came out which is exactly what we want to test for. We could make a bespoke enum that frames this in terms of components and modules, but that enum would essentially just be a reimplementation of `Cow`. However, I'll still try to take a look into making this clearer.